storageUnit/storageunit.go

```diff
@@ -259,14 +260,35 @@ func (u *Unit) DestroyUnit() error {
 	return u.persister.Destroy()
 }
 
+// ClearStorage will clean the storer by removing the database and creating a new one at the same location
+func (u *Unit) ClearStorage() error {
+	u.lock.Lock()
+	defer u.lock.Unlock()
+
+	err := u.persister.Close()
+	if err != nil {
+		return err
+	}
+
+	err = u.persister.DestroyClosed()
+	if err != nil {
+		return err
+	}
+
+	u.ClearCache()
+
+	u.persister, err = NewDB(u.argsNewDB)
+	return err
+}
+
 // IsInterfaceNil returns true if there is no value under the interface
 func (u *Unit) IsInterfaceNil() bool {
 	return u == nil
 }
 
 // NewStorageUnit is the constructor for the storage unit, creating a new storage unit
 // from the given cacher and persister.
-func NewStorageUnit(c types.Cacher, p types.Persister) (*Unit, error) {
+func NewStorageUnit(c types.Cacher, p types.Persister, argsNewDB ArgDB) (*Unit, error) {
```

Comment:
Because you did not implement the clear operation at the persister level, you need this argument which doesn't make sense whenever we need to create a new unit. A new unit should receive the persister and the cacher. The implementation should be don in the base level db component that has the path and the functionality for closing the db.

Reply:
done